### ISSUE TITLE
Option to disable polling

### ISSUE THREAD

**Issue by servo1x** [argoproj/argo-cd#1399]
**Is your feature request related to a problem? Please describe.**
We would like the option to disable git polling completely and as we only want to use the web hook for triggering updates.

**Describe the solution you'd like**
Option to disable git polling.

**Describe alternatives you've considered**
I know we can increase the polling interval, but it would be nice to just have the option to disable it completely.

**Comment by alexec** [comment]
Minor, have you tried to increase the interval to a very long time?

**Comment by anarcher** [comment]
IHMO, Disable polling gits is useful that CI pipeline takes git changes, building images and executes refresh-sync command like `argocd app refresh <app>` and `argocd app sync <app>`.  What do you think about this approach?

**Comment by alexec** [comment]
We'll be greatly reducing the number of requests in v1.1. Up to 5x (@alexmt to confirm).

With that in mind - do you still need this.

**Comment by alexmt** [comment]
Confirming, we are seeing 4x~5x fewer queries on our argocd instances. As we discussed offline, let collect data to find if implemented optimization is enough or we need to continue.

**Comment by alexec** [comment]
Have you tried to set the application controller parame `--app-resync`to a very high value?

You'd need to make the change here:

https://github.com/argoproj/argo-cd/blob/master/manifests/base/application-controller/argocd-application-controller-deployment.yaml#L26

**Comment by hfuss** [comment]
Re-posting my question here since this might be a good work around: https://github.com/argoproj/argo-cd/issues/3864#issuecomment-737939458

> We're experiencing something similar in v1.7.6.
>
> Our application controller started spewing this log this morning:
> ```
> patch: {"status":{"conditions":[{"lastTransitionTime":"2020-12-03T12:33:40Z","message":"rpc error: code = Unknown desc = Get \"https://github.com/Bandwidth/some-repo/info/refs?service=git-upload-pack\": dial tcp 140.82.113.3:443: i/o timeout (Client.Timeout exceeded while awaiting headers)","type":"ComparisonError"}]}}
> ---
> ```
> The reason being our network is saturated and so packets are being dropped, connections timing out, etc.
>
> So while this kind of behavior is expected during a network issue, I'm concerned as to the impact its having on application syncs.
>
> What bothers me, is that the repo server hasn't thrown a single error and everything it needs is still in its cache. So why is the app controller timing out on syncs and putting a bunch of apps in the Unknown state, when it has everything it needs to do its job?
>
> What is the point of the app controller hitting the Git repo with git-upload-pack?

**Comment by alexmt** [comment]
I think the root cause it the argocd tries to resolve latest revision every 3 minutes per application instead of per repository. Another issue caused by it is: https://github.com/argoproj/argo-cd/issues/5659

**Comment by moises-perez-tfs** [comment]
I believe #6102 solves something completely different from the original comment in this issue.

Does avoiding `timeout.reconciliation` disable polling?

**Comment by DanieleSassoli** [comment]
Or, rather than omitting `timeout.reconciliation`, maybe setting it to 0 should disable polling all together?

**Comment by alexmt** [comment]
I agree . The #6102 does not actually disable polling. This change ensures that Argo CD resolves revision only once per `timeout.reconciliation` ( before it was once per `timeout.reconciliation` * number of apps in repo ). So now you can disable polling by setting `timeout.reconciliation` to e.g. `30d`.

**Comment by alexmt** [comment]
I like your suggestion @DanieleSassoli . I think setting `timeout.reconciliation` to 0 should disable polling. 

@moises-perez-tfs , what do you think?

**Comment by moises-perez-tfs** [comment]
It sounds good to me.
